tutorials/introductory/animation_tutorial.py

```diff
@@ -0,0 +1,274 @@
+"""
+===========================
+Animations using matplotlib
+===========================
+
+Based on its plotting functionality, matplotlib also provides an interface to
+generate animations using the :class:`~matplotlib.animation` module. An
+animation is a sequence of frames where each frame corresponds to a plot on a
+:class:`~matplotlib.figure.Figure`. This tutorial covers a general guideline on
+how to create such animations and the different options available.
+"""
+
+import matplotlib.pyplot as plt
+import matplotlib.animation as animation
+import numpy as np
+
+###############################################################################
+# Animation Classes
+# =================
+#
+# The process of animation in matplotlib can be thought about in 2 different
+# ways:
+#
+# - :class:`~matplotlib.animation.FuncAnimation`: Generate data for first
+#   frame and then modify this data for each frame to create an animated plot.
+#
+# - :class:`~matplotlib.animation.FuncAnimation`: Generate a list (iterable)
+#   of artists that will draw in each frame in the animation.
+#
+# :class:`~matplotlib.animation.FuncAnimation` is more efficient in terms of
+# speed and memory as it draws an artist once and then modifies it. On the
+# other hand :class:`~matplotlib.animation.ArtistAnimation` is flexible as it
+# allows any iterable of artists to be animated in a sequence.
+#
+# :class:`~matplotlib.animation.FuncAnimation`
+# --------------------------------------------
+#
+# :class:`~matplotlib.animation.FuncAnimation` class allows us to create an
+# animation by passing a function that iteratively modifies the data of a plot.
+# This is achieved by using the *setter* methods on various
+# :class:`~matplotlib.artist.Artist`
+# (examples: :class:`~matplotlib.lines.Line2D`,
+# :class:`~matplotlib.collections.PathCollection`, etc.). A usual
+# :class:`~matplotlib.animation.FuncAnimation` object takes a
+# :class:`~matplotlib.figure.Figure` that we want to animate and a function
+# *func* that modifies the data plotted on the figure. It uses the *frames*
+# parameter to determine the length of the animation. The *interval* parameter
+# is used to determine time in milliseconds between drawing of two frames.
+# We will now look at some examples of using
+# :class:`~matplotlib.animation.FuncAnimation` with different artists.
+
+###############################################################################
+# Animating Lines
+# ^^^^^^^^^^^^^^^
+#
+# `.pyplot.plot` returns a :class:`~matplotlib.lines.Line2D` collection. The
+# data on this collection can be modified by using the
+# `.lines.Line2D.set_data` function. Therefore we can use this to modify the
+# plot using the function for every frame.
+
+fig, ax = plt.subplots()
+rng = np.random.default_rng()
+
+xdata = np.arange(0, 2 * np.pi, 0.01)
+(line,) = ax.plot(xdata, np.sin(xdata), c="b")
+ax.grid()
```

Comment:
```suggestion
```
`grid()` does not contribute to explaining animations. There's value in keeping the example code to the essential parts, so that users don't get distracted by unrelated details.